### ISSUE TITLE
Nonfunctioning PS2 keyboard connection

### ISSUE THREAD

**Issue by amishrabbit** [visrealm/pico-56#18]
Hey there. I built the PICO-56 kit that was sold through Hackerboxes. The unit has been flashed to the v 0.5 firmware and boots, but I have tested it with two known-functional PS/2 keyboards and the device does not seem to detect or recognize the keyboards, or there may be some other problem. The keytest.o applet cannot proceed because the device does not recognize any keys, nor do any of the lights on the keyboard function.

The Hackerboxes kit came with a classic NES gamepad, and that gamepad does function and is able to operate in the test apps.

How do I diagnose what is the root cause of this issue with the PS/2 keyboard or connector? I have closely inspected the solder joints on the PS/2 connector and they appear to be functional and not bridged. What other parts on the device could be at fault?

**Comment by visrealm** [comment]
Hi,
The keyboard is powered (grounded) via the transistor Q1. This transistor also has very fine pin pitch on the board and is very easy to bridge. Could you check the transistor connections?
When you turn the PICO-56 on, do the indicator lights on the keyboard flash once or not at all?
Cheers
Troy

**Comment by amishrabbit** [comment]
Hi Troy. When I power on the device, the blue SD indicator LED flashes once, the blue PWR LED on the board and the red PWR LED on the Pi remain lit steadily and do not flash. None of the LEDs on the keyboard flash at all.

I will inspect the Q1 transistor connections with a microscope; they look good but that's good info to check.

**Comment by visrealm** [comment]
Thanks. I was meaning the LEDs on the keyboard (Num lock, Caps lock, Scroll lock). They usually flash once at startup for most keyboards. If that's not happening, then it's likely a power issue (possibly Q1)

**Comment by Animapix** [comment]
Hi, I have the same issue with my keyboard. I checked all the soldering connections under a microscope, and they all seem to be good.
I'm using a USB ANSI keyboard with a US layout and a PS/2 adapter. I’m not sure if the issue is caused by the keyboard's power consumption or the USB/PS2 adapter.

**Comment by visrealm** [comment]
@Animapix I've tested mine with an old PS/2 keyboard which draws 400mA and it's fine.  I'd say it's the adapter. Is the keyboard designed to work with an adapter? The adapters will only work if the keyboard supports PS/2 protocol.

**Comment by Animapix** [comment]
Thank you for your response, @visrealm.

I tested the current consumption of my keyboard, which is 432 mA, so that seemed fine. However, as you pointed out, I checked and found that my keyboard is indeed not compatible with PS/2. I’m sorry for bothering you over something so simple! I'll order a compatible keyboard instead.

Thank you so much for all the work you’ve done on this impressive project. I'm continuing to explore how it works, and I must say, diving into retro computing has been quite a challenge, but also incredibly fascinating.

Thanks again!
